src/modules/user/middlewares/loggdeInUser.middleware.ts

```diff
@@ -0,0 +1,24 @@
+import { MedusaAuthenticatedRequest, MedusaMiddleware, Middleware } from 'medusa-extender';
+import { NextFunction, Response } from 'express';
+import UserService from '../../user/services/user.service';
+import * as jwt from 'jsonwebtoken';
+import { JwtPayload } from "jsonwebtoken";
+
+@Middleware({ requireAuth: true, routes: [{ method: "all", path: '*' }] })
+export class LoggedInUserMiddleware implements MedusaMiddleware {
+    public async consume(req: MedusaAuthenticatedRequest, res: Response, next: NextFunction): Promise<void> {
+        const userService = req.scope.resolve('userService') as UserService;
+        const userData: JwtPayload | string = jwt.decode(req?.session?.jwt);
```

Comment:
I wanted to update that part, could you remove everything related to jsonwebtoken and just use req.user.user_id ? It should be available

Reply:
Actually, the main issue I ran into while working on the tutorial was that req.user was not available. It always was undefined